### PR TITLE
Avoid swallowing exception information on cancelled error in REST channel

### PR DESCRIPTION
**Pull request by tmbo**
Includes the exception information in the logging output to allow for easier debugging of timeout related errors in the REST channel.

### CODE REVIEW

File: rasa/core/channels/rest.py [hunk 1]
```diff
@@ -166,7 +166,8 @@ async def receive(request: Request) -> Union[ResponseStream, HTTPResponse]:
                     )
                 except CancelledError:
                     logger.error(
-                        f"Message handling timed out for " f"user message '{text}'."
+                        f"Message handling timed out for user message '{text}'.",
+                        exec_info=True,
```
Comment (ancalita) [inline rasa/core/channels/rest.py:170]: Shouldn't this be `exc_info` parameter?

Comment (tmbo) [inline rasa/core/channels/rest.py:170]:
```suggestion
                        exc_info=True,
```

Comment (tmbo) [reply rasa/core/channels/rest.py]: thanks 🤦

**Review by ancalita — APPROVED** [review]
Even if it's a small change, should we also have a misc changelog entry?

**Pull request merged**